### DISCUSSION TITLE
Cannelloni für ESP32

### DISCUSSION THREAD

**Discussion by MyHomeMyData** [open3e/open3e#85] [Show and tell]
Es wurde an verschiedenen Stellen diskutiert, den Zugriff auf die CAN-Bus Daten mit einem ESP32 zu machen. Für die Verbindung zu Open3E bzw. zum ioBroker-Adapter bietet sich dabei ein TCP-Tunnel an. Auf Linux-Systemen wird dieser Ansatz im Cannelloni-Projekt realisiert. Entsprechende Projekte für den ESP32 gibt es auch, die haben bei mir aber nicht funktioniert.
 
Deshalb habe ich einen ESP32-basieter Cannelloni-Client entwickelt, der sich mit einem Cannelloni-Server verbindet und die Daten per TCP überträgt. Aufsetzen lässt sich der ESP32 mit vsCode PlatformIO oder mit der Arduino IDE.

Den aktuellen Status der laufenden Anwendung kann man sich mit einem simplen Web Interface anschauen:
![image](https://github.com/philippoo66/open3Edevelop/assets/144950531/940fdea5-6ecc-4e77-a9dc-04adacf876cc)

Programm und Readme gibt's [hier](https://github.com/MyHomeMyData/ESP32cannelloni).

Das ist ein Prototyp. Also gerne ausprobieren und hier Feedback geben.

Falls sich dieser Ansatz bewährt und genügend Interesse besteht, würde ich in den ioBroker-Adapter E3onCAN eine direkte IP-Schnittelle zum ESP32 einbauen, also den Umweg über Cannelloni und virtuellen CAN eliminieren. Dann würde der Adapter prinzipiell auch auf Windows laufen.
Bei Interesse an so einer Lösung bitte hier einen Kommentar einstellen.

**Comment by jwende** [comment]
Coole Geschichte - könnte man das parallel zu open3e am gleichen CAN Bus laufen lassen?

**Reply by MyHomeMyData** [reply]
Klar, Du kannst das parallel laufen lassen. Du solltest halt nicht aus 2 verschiedenen Quellen per UDSonCAN mit den gleichen ECU-Adressen reden. Mithören oder verschiedene ECU-Adressen sollten aber problemlos gehen.

**Reply by jwende** [reply]
Also muss ich mir einen Verteiler basteln ...
